src/sources/datadog/agent.rs

```diff
@@ -196,6 +216,13 @@ impl DatadogAgentSource {
             .or_else(|| header.map(Arc::from))
     }
 
+    fn is_valid_api_key(&self, api_key: &Option<Arc<str>>) -> bool {
+        if let (Some(k), Some(valid_api_keys)) = (api_key, &self.valid_api_keys) {
+            return valid_api_keys.contains(&k.to_string());
+        }
+        true
+    }
+
```

Comment:
Here's the function for validating the api_key. Does it make sense to accept any events if `self.valid_api_keys` is `None`? Also, if `self.valid_api_keys` is an empty vector and `self.drop_api_key` is true, all events will rejected.

Reply:
@prognant Just a friendly ping to see if you have any suggestion. Thanks!